### ISSUE TITLE
Writing through pointer of differently sized type

### ISSUE THREAD

**Issue by stilscher** [goblint/analyzer#582]
In the following example an integer is written through a char pointer. The following assert should fail, because only part of the integer value was overwritten. But currently the assert succeeds and this should be fixed.

https://github.com/goblint/analyzer/blob/59d7885d2dd62b451bc8db2a089d7a9550c3da2a/tests/regression/02-base/74-int-written-with-char-pointer.c#L1-L9

**Comment by michael-schwarz** [comment]
I guess while `//FAIL` is ideal, we'd be happy with `//UNKNOWN!` too here.

**Comment by michael-schwarz** [comment]
Yes, I think we can not just ignore the type of the left-hand side anymore and just go based off  the address.

**Comment by sim642** [comment]
This point came up again in a discussion with @vesalvojdani regarding all the things that addresses and their offsets, including #816. It seems that our addresses need to be typed.
